plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -87,8 +86,10 @@ def __init__(self, repo, publish_conduit, config, distributor_type, association_
         self.add_child(PublishCompsStep())
         self.add_child(PublishMetadataStep())
         self.add_child(CloseRepoMetadataStep())
-        self.add_child(GenerateSqliteForRepoStep(self.get_working_dir()))
         self.add_child(RemoveOldRepodataStep())
+        self.add_child(GenerateSqliteForRepoStep(self.get_working_dir()))
```

Comment:
What is the reason for moving that step?

Reply:
The reasoning was that `GenerateSqliteForRepoStep` is making changes to the metadata files that aren't tracked by the code that makes sure the `RemoveOldRepodataStep` doesn't touch them, so it feels safer to do it after the cleanup.  I'm 99% sure that it would still have been safe anyway because "old" is time-based and usually measured in days, but we probably shouldn't let it bypass the safety mechanism.